x/ibc/07-tendermint/types/consensus_state.go

```diff
@@ -13,22 +14,24 @@ import (
 
 // ConsensusState defines a Tendermint consensus state
 type ConsensusState struct {
-	Timestamp    time.Time               `json:"timestamp" yaml:"timestamp"`
-	Root         commitmentexported.Root `json:"root" yaml:"root"`
-	Height       uint64                  `json:"height" yaml:"height"`
-	ValidatorSet *tmtypes.ValidatorSet   `json:"validator_set" yaml:"validator_set"`
+	Timestamp          time.Time               `json:"timestamp" yaml:"timestamp"`
+	Root               commitmentexported.Root `json:"root" yaml:"root"`
+	Height             uint64                  `json:"height" yaml:"height"`
+	NextValidatorsHash tmbytes.HexBytes        `json:"next_validators_hash"` // validators hash for the next block
+	ValidatorSet       *tmtypes.ValidatorSet   `json:"validator_set" yaml:"validator_set"`
```

Comment:
I don't think we want to store the whole validator set in each consensus state - too large - can we store the hash and require that the validator set be passed in update messages instead?

Reply:
Hmm yea that's fine. Note this makes relayer flow for updating more complicated since the relayer needs to know the `LatestHeight` of the client in order to query the last trusted valset.

Though since this is true, we can just query for the `ValidatorSet` at `LatestHeight+1` so that we can use `NextValidators` of last trusted header in `lite.Verify`

Ref issue here: https://github.com/cosmos/ics/issues/456#issuecomment-665118922

Reply:
This requires significant changes with tradeoffs, I will implement this in a separate PR so people can see the diffs in isolation.

Reply:
opened an issue to track this #6930